docs/user-guide/references.rst

```diff
@@ -17,43 +17,85 @@ Glossary
+      Short for "effective area": it is the IRF representing the detector collection
+      area. See :ref:`irf-theory` and :ref:`irf-aeff`.
+
+    Bkg
+      Short for "background": it is the IRF representing the residual background rate
+      per solid angle. See :ref:`irf-theory` and :ref:`irf-bkg`.
 
     Cash
       The cash statistic is a Poisson fit statistic usually used when signal and
-      background can be modeled. It is defined as :math:`2 \times log(L)` See
+      background can be modeled [Cash1979]_. It is defined as :math:`2 \times log(L)`. See
       :ref:`cash` in :ref:`fit statistics <fit-statistics>`.
 
     Dataset
       In Gammapy a dataset bundles the data, IRFs, model and a likelihood function.
       Based on the model and IRFs the predicted number of counts are computed and
-      compared to the measured counts using the likelihood.
+      compared to the measured counts using the likelihood. See :ref:`datasets`
+
+    DL3
+      Short for "data level 3": it is used to mention or describe science-ready data, ie
+      the event list (with basically a Time, an arrival direction and an energy) and the
+      four IRFs (Aeff, EDisp, PSF, Bkg). See :ref:`data flow <data_flow>`.
+
+    DL4
+      Short for "data level 4": it is used to mention or describe binned-science data, ie
+      N-dim maps (multi-dimensional histograms) of the spatial, temporal and/or spectral
+      components of the DL3 data in instrumental units (e.g. counts). See :ref:`maps` and
+     :ref:`data flow <data_flow>`.
+
+    DL5
+      Short for "data level 5": it is used to mention or describe advanced-science data, ie
+      N-dim maps of the spatial, temporal and/or spectral astrophysical quantities in
+      physical units (e.g. cm-2 s-1). See :ref:`data flow <data_flow>`.
+
+    DL6
+      Short for "data level 6": it is used to mention or describe catalogs of sources,
+      with their spectral and spatial properties. See :ref:`data flow <data_flow>`.
 
     EDisp
-      Short for energy dispersion: it is the IRF that represents the probability
+      Short for "energy dispersion": it is the IRF that represents the probability
       of measuring a given reconstructed energy as a function of the true photon
-      energy. See :ref:`irf-edisp`
+      energy. See :ref:`irf-theory` and :ref:`irf-edisp`.
+
+    Estimator
+      Gammapy utility classes to compute astrophysical quantities based on a model.
+      They are used for the computation of DL5 products from the DL4 ones. See
+      :ref:`modeling` and the :ref:`data flow <data_flow>`.
 
     FoV
       Short for "field of view": it indicates the angular aperture (sometimes also the
       solid angle) on the sky that is visible by the instrument with a single pointing.
 
+    FoV Background
+      Background estimation method typically used for the 3D analysis. See :ref:`makers`.
+
+    HLI
+      Short for "high level interface: this Python API aims to realize most of the analysis
```

Comment:
Maybe python is obvious here, no?
`this API supports a large fraction of analysis use cases using a simple interface and YAML configuration files.`